src/filters/frame_skipping_pp.py

```diff
@@ -0,0 +1,52 @@
+from src.filters.abstract_pp import AbstractPP
+from src.utils import image_utils
+from typing import List
+
+
+class frameSkippingPP(AbstractPP):
+
+    def __init__(self, threshold=0.0, compare_foreground=False, 
+                 distance_metric='absolute_difference'):
+        self.threshold = threshold
+        self.compare_foreground = compare_foreground
+        self.distance_metric = distance_metric
+
+    def predict(self, batch) -> List[bool]:
+        prev_frame = None
+        skipFrames = []
+        for frame_batch in batch:
+            frame = frame_batch.frames_as_numpy_array()[0]
+            if prev_frame is not None:
+                """
+                If compare_foreground set to true, calculate distance
+                metric on only the foreground pixels
+                else on the entire image.
+                """
+                if self.compare_foreground is True:
+                    frame_diff = self.compare_foreground_mask(
+                        frame, prev_frame, self.distance_metric)
+                else: 
+                    frame_diff = self.frame_difference(
+                        frame, prev_frame, self.distance_metric)
+                if frame_diff < self.threshold:
+                    skipFrames.append(True)
+                else:
+                    skipFrames.append(False)
+                    prev_frame = frame
+            else:
+                skipFrames.append(False)
+                prev_frame = frame
+        return skipFrames
+
+    # Function to calculate frame difference based on distance metric
```

Comment:
use docstring to write comments

Reply:
Also, please add type hints both for parameters and return type.

Reply:
Added type hints and converted all comments to docstring